File: Daily/C#/Leetcode567-字符串的排列.cs
public class Solution {
    public bool CheckInclusion(string s1, string s2) {
        int patternsize = s1.Length;
        int n = s2.Length;
        if(patternsize > n){
            return false;
        }
        int[] nums1 = new int[26];
        int[] nums2 = new int[26];
        for(int i = 0;i < patternsize;i++){
            nums1[s1[i] - 'a']++;
            nums2[s2[i] - 'a']++;
        }
        if(compare(nums1,nums2)){
            return true;
        }
        for(int i = patternsize;i < n;i++){
            nums2[s2[i] - 'a']++;
            nums2[s2[i - patternsize] - 'a']--; 
            if(compare(nums1,nums2)){
                return true;
            }
        }       
        return false;
    }
    public bool compare(int[] nums1,int[] nums2){
        if(nums1 == null || nums2 == null || nums1.Length != nums2.Length){
            return false;
        }
        for(int i = 0;i < nums1.Length;i++){
            if(nums1[i] != nums2[i]){
                return false;
            }
        }
        return true;
    }
}

File: Daily/Kotlin/Leetcode567-字符串的排列.kt
class Solution {
    fun checkInclusion(s1: String, s2: String): Boolean {
        val patternsize =  s1.length
        val n = s2.length
        if(n < patternsize){
            return false
        }
        var nums1 = IntArray(26)
        var nums2 = IntArray(26)
        for(i in 0 until patternsize){
            nums1[s1[i] - 'a']++
            nums2[s2[i] - 'a']++
        }
        if(compare(nums1,nums2)){
            return true
        }
        for(i in patternsize until n){
            nums2[s2[i] - 'a']++
            nums2[s2[i-patternsize] - 'a']--
            if(compare(nums1,nums2)){
                return true 
            }
        }
        return false
    }
    fun compare(nums1:IntArray,nums2:IntArray):Boolean{
        if(nums1.size != nums2.size || nums1 == null || nums2 == null){
            return false
        }
        for(i in 0 until nums1.size){
            if (nums1[i] != nums2[i]){
                return false
            }
        }
        return true
    }
}